track_oracle/core/kwiver_io_base.txx

```diff
@@ -30,10 +30,18 @@ typedef char (&no_io_type)[2];
 typedef char yes_io_type;
 
 struct proxy{ template <typename U> proxy(const U&); };
+# if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable : 4172 ) /* returning address of local variable */
```

Comment:
Off the top of my head, I'm not sure why `no_io_type` is a reference. It's certainly not what I would've tried first. Everywhere else in track_oracle, I think, yes-types are a `char` and no-types are something whose `sizeof()` is guaranteed to be different than yes-type, usually `char[2]`. So if it's a reference here, it was to make something work, but the details escape me at the moment.

That said, I'm certain it would all benefit from a refactoring, which I would love to do when/if I get a chance.

(The background in case anybody cares: track_oracle reads and writes types to both XML and CSV files, see `kwiver_io_base.txx`. Ideally this would also extend to e.g. json in the future. Whenever possible, it wants to use the same method for both: `from_str()` for reading and `to_stream()` for writing. Again whenever possible, it wants to use the type's native `op>>` and `op<<`, so that when you add a type, everything 'just works' and no extra code needs to be written to support the type. **However**, and here's where my memory starts to get fuzzy, the XML reading uses tinyxml, which ends up being string-token based (which is why it reads from a string but outputs to a stream.) But this has problems if the `op>>` representation contains spaces. (For example, vidtk's old timestamp class.) So another layer of indirection is added, defining `kwiver_read` and `kwiver_write` to either use our local space-less methods (if defined in  `kwiver_io_helpers.h`) or the type's `op>>` and `op<<` (if not defined in `kwiver_io_helpers.h`.))

Like I said, I'd love to refactor it, both to simplify this and to make the functionality more generally available in kwiver.